server/src/main/java/org/apache/druid/server/coordinator/helper/NewestSegmentFirstIterator.java

```diff
@@ -458,9 +401,14 @@ String getDataSource()
       this.segments = segments;
     }
 
-    public List<DataSegment> getSegments()
+    List<DataSegment> getSegments()
     {
       return segments;
     }
+
+    int size()
```

Comment:
can we call this getSize() and explicitly list if it is public/private/protected?

Reply:
Done.